app/domain/authentication/authn_azure/authenticator.rb

```diff
@@ -6,35 +6,77 @@ module AuthnAzure
     Err = Errors::Authentication::AuthnAzure
     Log = LogMessages::Authentication::AuthnAzure
     # Possible Errors Raised:
-    # TODO: Add errors
+    # TokenFieldNotFoundOrEmpty, MissingRequestParam
 
     Authenticator = CommandClass.new(
       dependencies: {
-        fetch_authenticator_secrets: Authentication::Util::FetchAuthenticatorSecrets.new,
-        logger:                      Rails.logger
+        fetch_authenticator_secrets:   Authentication::Util::FetchAuthenticatorSecrets.new,
+        verify_and_decode_token:       Authentication::AuthnOidc::VerifyAndDecodeToken.new,
```

Comment:
using OIDC as a dependency will also enforce us to use the OIDC specific errors and logs
Ex: `OIDCProviderCertificateFetchedFromCache` , etc